### PR TITLE
[Merged by Bors] - refactor(CategoryTheory/Adjunction): make uniqueness of adjoints independent of opposites

### PR DESCRIPTION
**Pull request by dagurtomas**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by dagurtomas** [comment]
!bench

**Comment by dagurtomas** [comment]
!bench

File: Mathlib/CategoryTheory/Adjunction/Unique.lean [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+@[reassoc (attr := simp)]
+theorem leftAdjointUniq_hom_app_counit {F F' : C ⥤ D} {G : D ⥤ C} (adj1 : F ⊣ G) (adj2 : F' ⊣ G)
+    (x : D) :
+    (leftAdjointUniq adj1 adj2).hom.app (G.obj x) ≫ adj2.counit.app x = adj1.counit.app x := by
+  rw [← leftAdjointUniq_hom_counit adj1 adj2]
+  rfl
+#align category_theory.adjunction.left_adjoint_uniq_hom_app_counit CategoryTheory.Adjunction.leftAdjointUniq_hom_app_counit
+
+@[simp]
```
Comment (joelriou) [inline Mathlib/CategoryTheory/Adjunction/Unique.lean:155]: I would prefer if this lemma was not a `simp` lemma:
```suggestion
```

File: Mathlib/CategoryTheory/Adjunction/Unique.lean [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+#align category_theory.adjunction.right_adjoint_uniq_hom_app_counit CategoryTheory.Adjunction.rightAdjointUniq_hom_app_counit
+
+@[reassoc (attr := simp)]
+theorem rightAdjointUniq_hom_counit {F : C ⥤ D} {G G' : D ⥤ C} (adj1 : F ⊣ G) (adj2 : F ⊣ G') :
+    whiskerRight (rightAdjointUniq adj1 adj2).hom F ≫ adj2.counit = adj1.counit := by
+  ext
+  simp
+#align category_theory.adjunction.right_adjoint_uniq_hom_counit CategoryTheory.Adjunction.rightAdjointUniq_hom_counit
+
+@[simp]
```
Comment (joelriou) [inline Mathlib/CategoryTheory/Adjunction/Unique.lean:227]:
```suggestion
```

**Review by joelriou — APPROVED** [review]

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**